app/api/services/evidence.py

```diff
@@ -83,6 +84,54 @@ def get_previous_submitted_evidence_for_supplier_and_domain(self, evidence_id, d
         evidence = query.first()
         return evidence
 
+    def get_evidence_data(self, evidence_id):
+        evidence = {}
+        get_evidence_by_id_query = (
+            db.session.query(Evidence)
+            .filter(
+                Evidence.id == evidence_id
+            )
+            .options(
+                joinedload(Evidence.supplier).joinedload(Supplier.signed_agreements),
+                joinedload(Evidence.supplier).joinedload(Supplier.domains),
+                joinedload(Evidence.user),
+                joinedload(Evidence.brief).joinedload(Brief.clarification_questions),
+                joinedload(Evidence.brief).joinedload(Brief.work_order),
+                joinedload(Evidence.domain),
+                raiseload('*')
+            )
+        )
+
+        value = get_evidence_by_id_query.one_or_none()
+        evidence = value.serialize()
+        evidence['domain_name'] = value.domain.name
+
+        evidence_subquery = (
+            db
+            .session
+            .query(
+                func.json_array_elements_text(Evidence.data['criteria']).label('domain_criteria_id')
+            )
+            .filter(Evidence.id == evidence_id)
+            .subquery()
+        )
+
+        query = (
```

Comment:
I think we can make improvements to how this data is returned for this query.  Does this query give you all the data that you need for the page?
```SQL
select e.id, e.data, q2.domain_criteria
from evidence e
inner join (
	select 
		q1.id evidence_id,
		json_object_agg(
			q1.domain_criteria_id,
			json_build_object(
				'name', q1.name
			)
		) domain_criteria
	from (
		select e.id, domain_criteria_id, dc.name 
		from evidence e, json_array_elements_text(e.data -> 'criteria') domain_criteria_id
		inner join domain_criteria dc on dc.id = domain_criteria_id::integer
	) q1
	group by q1.id
) q2 on q2.evidence_id = e.id
```